src/components/CertificateViewer.tsx

```diff
@@ -34,16 +34,31 @@ export const CertificateViewer = ({
   handleSendCertificate,
 }: CertificateViewerProps) => {
   const tokenRegistryAddress = getTokenRegistryAddress(document);
-  const [templates, setTemplates] = useState([]);
-  const [selectedTemplate, setSelectedTemplate] = useState("");
+  const [templates, setTemplates] = useState<{ id: string; label: string; type: string }[]>([]);
+  const [selectedTemplate, setSelectedTemplate] = useState<string>("");
   const originalData = getData(document);
   const attachments = originalData?.attachments;
   const hasAttachments = attachments && attachments.length > 0;
 
-  const updateTemplates = useCallback((templates) => {
-    setTemplates(templates);
-    setSelectedTemplate(templates[0].id);
-  }, []);
+  const updateTemplates = useCallback(
+    (templates) => {
+      setTemplates(templates);
+      setSelectedTemplate(templates[0].id);
+
+      // modify tabs by reseting templates if attachment(s) exists
+      if (hasAttachments) {
```

Comment:
Why not just running that branch always? I don't understand the need to check if there are attachments it did the same with a filter, just always run the filter